nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/WildcardIndexPath.java

```diff
@@ -65,6 +65,7 @@
                 } else {
                     final Object[] array = (Object[]) value;
                     return IntStream.range(0, array.length)
+                        .filter(index -> array[index] != null)
```

Comment:
@pvillard31 I don't think this is exactly right. The wildcard should include all elements, including nulls. Imagine, for example, that I have some data like:

```
{
  "id" : 3,
  "numbers" : [ 1, null, 4 ]
}

```
If I want to set /numbers[*] to the value 8, then I would expect to get back:

```
{
  "id" : 3,
  "numbers" : [ 8, 8, 8 ]
}
```
But with this modification I'd get back
```
{
  "id" : 3,
  "numbers" : [ 8, null, 8 ]
}
```

Reply:
Hey @markap14, thanks for the review. While I agree with your comment, it turns out (if I looked correctly) that, currently, updating an array of simple elements is not possible. If I set:
```
/numbers[*]
````
as the path to update with a value of 8, it won't update anything.

I rebased against master and pushed another commit to fix the update action when updating an array of simple (not records) elements, and also to address your comment. Let me know if it looks like the right approach to you.